### PR TITLE
Adds a migration to remove _id from filter items

### LINKED ISSUE #3485: Settings/Filters update throw a validation error when the instance has grouped filters
This is happening in instances that have grouped filters, created some time ago. Not possible to reproduce with a fresh instance (at least with the tested cases).
The object sent to the API contains an additional item property that's not allowed `_id` that contains the value of the property id.
The error occurs at saving Settings of Filters even without any changes.

To solve it's necessary to do a migration that removes this property.
Screen Shot 2021-02-26 at 10 46 38

### PR DESCRIPTION
**Pull request by grafitto**
fixes #3485 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [x] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/migrations/migrations/34-remove-_id-from-filter-items/specs/34-remove-_id-from-filter-items.spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import testingDB from 'api/utils/testing_db';
+import migration from '../index.js';
+import fixtures from './fixtures.js';
+
+describe('migration remove-_id-from-filter-items', () => {
+  beforeEach(async () => {
+    spyOn(process.stdout, 'write');
+    await testingDB.clearAllAndLoad(fixtures);
```
Comment (RafaPolit) [inline app/api/migrations/migrations/34-remove-_id-from-filter-items/specs/34-remove-_id-from-filter-items.spec.js:8]: The migration looks good, and I believe it works as expected.

Still, I am concerned that the fixtures only show a single scenario: "A collection whose settings have filters, and those filters have nested items with _id present".

There are many other scenarios that need to be tested that they are not affected:
- Collections that don't have filters
- Collections that have filters and don't have nested items
- Collections that have filters with nested items and no _id present in there.

**Review by RafaPolit — CHANGES_REQUESTED** [review]

**Comment by RafaPolit** [comment]
While this migration may be needed, the fact that we have no clue as how to those _id's appeared there may leave us having to re-run an identical migration if this condition happens again. Please, spend some time trying to learn how this condition appeared in the first place, in order to first fix the problem, and only then run the migration.

File: app/api/migrations/migrations/34-remove-_id-from-filter-items/specs/34-remove-_id-from-filter-items.spec.js (hunk 1 above)
Comment (grafitto) [reply app/api/migrations/migrations/34-remove-_id-from-filter-items/specs/34-remove-_id-from-filter-items.spec.js:8]: I have added a few test cases as suggested, you can check them out when you get time

**Comment by grafitto** [comment]
> While this migration may be needed, the fact that we have no clue as how to those _id's appeared there may leave us having to re-run an identical migration if this condition happens again. Please, spend some time trying to learn how this condition appeared in the first place, in order to first fix the problem, and only then run the migration.

I have tried creating new filters and grouped filters, both do not add `_id` into the database, There is a high chance we are dealing with old data, maybe before the decision was made to remove the field from the database. However I am not sure whether there are other cases where filters are creating other than the settings > filters section. If so you can let me know so that I check it out.

**Pull request merged**